driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -72,7 +76,12 @@ public static IntegerExpression of(final long of) {
     public static NumberExpression of(final double of) {
         return new MqlExpression<>((codecRegistry) -> new AstPlaceholder(new BsonDouble(of)));
     }
+    public static NumberExpression of(final BigDecimal of) {
```

Comment:
Note that `BigDecimal` is not equivalent to `Decimal128` (the IEEE 754 quadruple-precision binary floating-point format). We need to support `Decimal128` because MongoDB can do computations with values of this type.

Reply:
+1